src/app.ts

```diff
@@ -13,14 +13,38 @@ interface TemperatureSummary {
   average: number
 }
 
+const dataByCity: { [key: string]: TemperatureReading[] } = {}
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  readings.forEach((reading: TemperatureReading): void => {
```

Comment:
Why the usage of `forEach` instead of other array method?

Reply:
I find the syntax a bit more readable, when iterating over the array I can extract and push elements into the global array by accessing the object properties and also check conditionally before adding the data.